test/integration/scheduler_perf/util.go

```diff
@@ -44,63 +37,25 @@ const enableEquivalenceCache = true
 // remove resources after finished.
 // Notes on rate limiter:
 //   - client rate limit is set to 5000.
-func mustSetupScheduler() (schedulerConfigurator scheduler.Configurator, destroyFunc func()) {
-
-	h := &framework.MasterHolder{Initialized: make(chan struct{})}
-	s := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
-		<-h.Initialized
-		h.M.GenericAPIServer.Handler.ServeHTTP(w, req)
-	}))
-
-	framework.RunAMasterUsingServer(framework.NewIntegrationTestMasterConfig(), s, h)
-
+func mustSetupScheduler() (scheduler.Configurator, perfutil.ShutdownFunc) {
+	masterURL, masterShutdown := perfutil.StartTestMaster()
 	clientSet := clientset.NewForConfigOrDie(&restclient.Config{
-		Host:          s.URL,
+		Host:          masterURL,
 		ContentConfig: restclient.ContentConfig{GroupVersion: testapi.Groups[v1.GroupName].GroupVersion()},
 		QPS:           5000.0,
 		Burst:         5000,
 	})
 
-	informerFactory := informers.NewSharedInformerFactory(clientSet, 0)
-
-	schedulerConfigurator = factory.NewConfigFactory(
-		v1.DefaultSchedulerName,
-		clientSet,
-		informerFactory.Core().V1().Nodes(),
-		informerFactory.Core().V1().Pods(),
-		informerFactory.Core().V1().PersistentVolumes(),
-		informerFactory.Core().V1().PersistentVolumeClaims(),
-		informerFactory.Core().V1().ReplicationControllers(),
-		informerFactory.Extensions().V1beta1().ReplicaSets(),
-		informerFactory.Apps().V1beta1().StatefulSets(),
-		informerFactory.Core().V1().Services(),
-		informerFactory.Policy().V1beta1().PodDisruptionBudgets(),
-		informerFactory.Storage().V1().StorageClasses(),
-		v1.DefaultHardPodAffinitySymmetricWeight,
-		enableEquivalenceCache,
-	)
-
-	eventBroadcaster := record.NewBroadcaster()
-	eventBroadcaster.StartRecordingToSink(&clientv1core.EventSinkImpl{Interface: clientv1core.New(clientSet.CoreV1().RESTClient()).Events("")})
-
-	sched, err := scheduler.NewFromConfigurator(schedulerConfigurator, func(conf *scheduler.Config) {
-		conf.Recorder = eventBroadcaster.NewRecorder(legacyscheme.Scheme, v1.EventSource{Component: "scheduler"})
-	})
-	if err != nil {
-		glog.Fatalf("Error creating scheduler: %v", err)
-	}
-
-	stop := make(chan struct{})
-	informerFactory.Start(stop)
+	evtBroadcaster := record.NewBroadcaster()
```

Comment:
I believe this and below variables can move inside `StartScheduler()`?

Reply:
So the thinking here is that the test might want to setup the appropriate environment and then have the scheduler (or in further other modules) start in that environment. That is why I abstracted it out from the StartScheduler() code. If the eventBroadcaster is scheduler specific, I can move it in, but for the clientSet, I believe it might be better for it to be outside as it can be configured differently based on different testing needs. @shyamjvs what do you think?

Reply:
Yes, for clientset I agree it makes sense to offer the flexibility (wrt qps, etc). For event-broadcaster I'm not very sure if that'd be useful - and probably just adds complexity to the utility function.